engine/src/test/java/org/camunda/bpm/engine/test/api/history/removaltime/cleanup/HistoryCleanupByteArrayRemovalTest.java

```diff
@@ -0,0 +1,179 @@
+  }
+
+  @After
+  public void tearDown() {
+    restoreCleanupJobHandler();
+    runHistoryCleanup();
+  }
+
+  @Test
+  @RemoveAfter
+  public void shouldHaveRemovalTimeOnFailingHistoryCleanupJob() {
+    // given
+    engineConfiguration.setHistoryCleanupJobLogTimeToLive("1");
+    overrideFailingCleanupJobHandler();
+
+    try {
+      // when
+      runHistoryCleanup();
+      fail("This test should fail during history cleanup and not reach this point");
+    } catch (Exception e) {
+      HistoricJobLogEvent event = getLastFailingHistoryCleanupJobEvent();
+      String exceptionByteArrayId = event.getExceptionByteArrayId();
+      ByteArrayEntity byteArray = findByteArrayById(exceptionByteArrayId);
+
+      // then
+      assertThat(byteArray).isNotNull();
+      assertThat(byteArray.getRemovalTime()).isNotNull();
+    }
+  }
+
+  protected ByteArrayEntity findByteArrayById(String byteArrayId) {
+    return engineConfiguration.getCommandExecutorTxRequired().execute(new GetByteArrayCommand(byteArrayId));
+  }
+
+  protected void restoreCleanupJobHandler() {
+    engineConfiguration.getJobHandlers().put(HistoryCleanupJobHandler.TYPE, new HistoryCleanupJobHandler());
+  }
+
+  protected void overrideFailingCleanupJobHandler() {
+    engineConfiguration.getJobHandlers().put(HistoryCleanupJobHandler.TYPE, new FailingHistoryCleanupJobHandler());
+  }
+
+  protected List<Job> runHistoryCleanup() {
+    historyService.cleanUpHistoryAsync(true);
+
+    List<Job> jobs = historyService.findHistoryCleanupJobs();
+    List<String> jobIds = new ArrayList<>();
+
+    for (Job job : jobs) {
+      jobIds.add(job.getId());
+      managementService.executeJob(job.getId());
+    }
+
+    return jobs;
+  }
+
+  protected HistoricJobLogEvent getLastFailingHistoryCleanupJobEvent() {
+    return (HistoricJobLogEvent) historyService.createHistoricJobLogQuery()
+        .failureLog()
+        .jobDefinitionType("history-cleanup")
+        .singleResult();
+  }
+
+  protected void initEngineConfiguration() {
+    engineConfiguration = engineRule.getProcessEngineConfiguration();
+
+    engineConfiguration.setHistoryRemovalTimeStrategy(HISTORY_REMOVAL_TIME_STRATEGY_END)
+        .setHistoryRemovalTimeProvider(new DefaultHistoryRemovalTimeProvider())
+        .initHistoryRemovalTime();
+
+    engineConfiguration.setHistoryCleanupStrategy(HISTORY_CLEANUP_STRATEGY_REMOVAL_TIME_BASED);
+
+    engineConfiguration.setHistoryCleanupBatchSize(MAX_BATCH_SIZE);
+    engineConfiguration.setHistoryCleanupBatchWindowStartTime(null);
+    engineConfiguration.setHistoryCleanupDegreeOfParallelism(1);
+
+    engineConfiguration.setBatchOperationHistoryTimeToLive(null);
+    engineConfiguration.setBatchOperationsForHistoryCleanup(null);
+
+    engineConfiguration.setHistoryTimeToLive(null);
+
+    engineConfiguration.setTaskMetricsEnabled(false);
+    engineConfiguration.setTaskMetricsTimeToLive(null);
+
+    engineConfiguration.initHistoryCleanup();
+  }
```

Comment:
❌ Tests that use the `ProvidedProcessEngineRule` should either not adjust `ProcessEngineConfiguration` or return to the values the test class encountered upon initialization. Otherwise, you might influence upcoming test classes. The engine and its configuration are shared between all tests using the `ProvidedProcessEngineRule` to reduce ramp-up time and resources for each test, i.e. by reusing the process engine. Another way of dealing with adjusted configuration is by using the [ProcessEngineBootstrapRule](https://github.com/camunda/camunda-bpm-platform/blob/master/engine/src/test/java/org/camunda/bpm/engine/test/util/ProcessEngineBootstrapRule.java). You can search for usages of this class to see how it's used in other tests.

Reply:
Done 👌